Sources/SwiftFormat/Rules/AlwaysUseLiteralForEmptyArrayInit.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftParser
+
+/// Never use `[<Type>]()` syntax. In call sites that should be replaced with `[]`,
+/// for initializations use explicit type combined with empty array literal `let _: [<Type>] = []`
+/// Static properties of a type that return that type should not include a reference to their type.
+///
+/// Lint:  Non-literal empty array initialization will yield a lint error.
+/// Format: All invalid use sites would be related with empty literal (with or without explicit type annotation).
+@_spi(Rules)
+public final class AlwaysUseLiteralForEmptyArrayInit : SyntaxFormatRule {
```

Comment:
Can you add support for dictionaries as well? The rule could be named `AlwaysUseLiteralForEmptyCollections` (I don't think we need `Init` in the name).

Reply:
I can look into that sure!

Reply:
Done!